queue_services/payment-reconciliations/src/reconciliations/eft/eft_base.py

```diff
@@ -0,0 +1,104 @@
+# Copyright © 2023 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This manages the EFT base class."""
+import decimal
+from datetime import datetime
+
+from reconciliations.eft.eft_enums import EFTConstants
+from reconciliations.eft.eft_errors import EFTError
+from reconciliations.eft.eft_parse_error import EFTParseError
+
+
+class EFTBase:
+    """Defines the structure of the base class of an EFT record."""
+
+    record_type: str  # Always 1 for header, 2 for transaction, 7 for trailer
+    content: str
+    index: int
+    errors: [EFTParseError]
+
+    def __init__(self, content: str, index: int):
+        """Return an EFT Base record."""
+        self.content = content
+        self.index = index
+        self.errors = []
+
+    def is_valid_length(self, length: int = EFTConstants.EXPECTED_LINE_LENGTH.value) -> bool:
+        """Validate content is the expected length."""
+        if self.content is None or len(self.content) != length:
+            return False
+
+        return True
+
+    def validate_record_type(self, expected_record_type: str) -> bool:
+        """Validate if the record type is the expected value."""
+        if not self.record_type == expected_record_type:
+            self.add_error(EFTParseError(EFTError.INVALID_RECORD_TYPE, self.index))
+
+    def parse_decimal(self, value: str, error: EFTError) -> decimal:
+        """Try to parse decimal value from a string, return None if it fails and add an error."""
+        try:
+            # ends with blank or minus sign, handle the minus sign situation
+            if value.endswith('-'):
+                value = '-' + value[:-1]
+
+            result = decimal.Decimal(value)
```

Comment:
should probably add str(value) just in case here

Reply:
EG.

Also Decimal(10.00) works differently than Decimal('10.00')

Reply:
Thanks for the insight. Updated.